### ISSUE TITLE
TSServer: config file diagnostics event not sent if config file changes

### ISSUE THREAD

**Issue by dbaeumer** [Microsoft/TypeScript#10473]
**TypeScript Version:**  2.0.0
**Code**
_tsconfig.json_

``` json
{
    "compilerOptions": {
        "module": "commonjs",
        "aaa": 10
    }
}
```

The server sends and event 

```
Event received: configFileDiag (0).
Data: {
    "triggerFile": "p:/mseng/VSCode/Playgrounds/ts-2.0/test.ts",
    "configFile": "p:/mseng/VSCode/Playgrounds/ts-2.0/tsconfig.json",
    "diagnostics": [
        {
            "text": "Unknown compiler option 'aaa'."
        }
    ]
}
```

which is correct. Fix the error in the tsconfig.json and save.

Observe: no configFileDiag update is sent. We either need an event to reqeuest configFileDiag or it should automatically happen when the tsconfig.json file is saved and recognized by the server.

**Comment by mhegazy** [comment]
should be handled by https://github.com/Microsoft/TypeScript/pull/10418

**Comment by mhegazy** [comment]
the fix should be in the nightly today

**Comment by dbaeumer** [comment]
@mhegazy I tested this today with TS 2.0.3 and this is still not working. An error is reported once and even if I change and save the tsconfig.json the errors are not rechecked.

**Comment by dbaeumer** [comment]
@mhegazy would it be possible to fix this for 2.0.5

**Comment by mhegazy** [comment]
@zhengbli can you take a look? this should have been fixed already?

**Comment by dbaeumer** [comment]
@mhegazy sorry for the swirl. Made a mistake during debugging. Will check and and reopen if it is really still an issue.

**Comment by dbaeumer** [comment]
OK. Now I see. The triggerFile is in that case tsconfig.json which I need to handle differently.

**Comment by dbaeumer** [comment]
@mhegazy @zhengbli there is still a problem. When an error exist, you fix it and save the tsconfig.json file no notification is sent that no errors exists anymore. The server only sends if errors exists. To clear the error list I need a notification in the no error case as well.

**Comment by zhengbli** [comment]
I have a fix locally, will send out an PR once the merge between master and release-2.0.5 at https://github.com/Microsoft/TypeScript/pull/11197 is done so I don't need to repeat that effort.

**Comment by dbaeumer** [comment]
Please see: https://github.com/Microsoft/TypeScript/issues/11615

**Comment by dbaeumer** [comment]
When will this land in 2.0.6 ?

**Comment by zhengbli** [comment]
This should go with the next release, as it has already been ported to the release branch (release-2.0.5). @mhegazy may know better about the release schedule.

**Comment by mhegazy** [comment]
We will publish an insider drop today.

**Comment by dbaeumer** [comment]
I tested this again on 2.0.9 and this is still not working. Here is what I do:

- create a tsconfig.json with an invalid property
- open a TS file which activates the tsconfig.json
- you correctly get an error
- change the wrong property to another invalid property name

Observe: you don't get an event informing you about the new invalid property name

- change the property name to a valid property name

Observe: you don't get an event informing you that no error exists

**Comment by dbaeumer** [comment]
@mhegazy @zhengbli can you please have a look

**Comment by dbaeumer** [comment]
If you close and reopen the TS file then the tsconfig.json gets revalidated.

**Comment by zhengbli** [comment]
This should be fixed in master now, @dbaeumer can you give it a try? thanks

**Comment by alexlur** [comment]
Is this supposed to land in 2.1?

**Comment by dbaeumer** [comment]
@mjbvz can you have a look. To test it when this lands in 2.1.3 you need to uncomment the code here: https://github.com/Microsoft/vscode/blob/master/extensions/typescript/src/typescriptMain.ts#L374
